app/src/main/java/com/example/bookwormadventuresdeluxe2/BorrowDetailViewFragment.java

```diff
@@ -160,9 +167,25 @@ private void btnRequestBook(View view)
         this.bookDocument.update(getString(R.string.requesters),
                 FieldValue.arrayUnion(UserCredentialAPI.getInstance().getUsername()));
         this.bookDocument.update(getString(R.string.status), getString(R.string.requested));
+
+        // Send In-app and Push notification to owner
+        sendBorrowRequestNotification();
         onBackClick(view);
     }
 
+    private void sendBorrowRequestNotification()
+    {
+        /* Create notification for firestore collection */
+        String message = "New borrow request from: "
+                + UserCredentialAPI.getInstance().getUsername();
+        HashMap<String, String> inAppNotification = new HashMap<>();
+        inAppNotification.put("bookID", selectedBookId);
+        inAppNotification.put("message", message);
+        inAppNotification.put("timestamp", String.valueOf(Timestamp.now().getSeconds()));
+        /* Call notification handler to process notification */
+        NotificationHandler.sendNotification("Borrow Request", message, selectedBook.getOwner(), inAppNotification);
+    }
+
     private void btnSetLocation(View view)
```

Comment:
Would you mind adding a Javadoc header to this too?

Reply:
done